native/src/main/java/io/ballerina/lib/aws/mpe/AwsMpeClient.java

```diff
@@ -0,0 +1,130 @@
+import software.amazon.awssdk.auth.credentials.AwsBasicCredentials;
+import software.amazon.awssdk.auth.credentials.AwsCredentials;
+import software.amazon.awssdk.auth.credentials.AwsCredentialsProvider;
+import software.amazon.awssdk.auth.credentials.AwsSessionCredentials;
+import software.amazon.awssdk.auth.credentials.StaticCredentialsProvider;
+import software.amazon.awssdk.services.marketplaceentitlement.MarketplaceEntitlementClient;
+import software.amazon.awssdk.services.marketplaceentitlement.model.GetEntitlementsRequest;
+import software.amazon.awssdk.services.marketplaceentitlement.model.GetEntitlementsResponse;
+
+import java.util.Objects;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+
+/**
+ * Representation of {@link software.amazon.awssdk.services.marketplaceentitlement.MarketplaceEntitlementClient} with
+ * utility methods to invoke as inter-op functions.
+ */
+public final class AwsMpeClient {
+    private static final String NATIVE_CLIENT = "nativeClient";
+    private static final ExecutorService EXECUTOR_SERVICE = Executors.newCachedThreadPool(new AwsMpeThreadFactory());
+
+    private AwsMpeClient() {
+    }
+
+    /**
+     * Creates an AWS MPE native client with the provided configurations.
+     *
+     * @param bAwsMpeClient The Ballerina AWS MPE client object.
+     * @param configurations AWS MPE client connection configurations.
+     * @return A Ballerina `mpe:Error` if failed to initialize the native client with the provided configurations.
+     */
+    public static Object init(BObject bAwsMpeClient, BMap<BString, Object> configurations) {
+        try {
+            ConnectionConfig connectionConfig = new ConnectionConfig(configurations);
+            AwsCredentials credentials = getCredentials(connectionConfig);
+            AwsCredentialsProvider credentialsProvider = StaticCredentialsProvider.create(credentials);
+            MarketplaceEntitlementClient nativeClient = MarketplaceEntitlementClient.builder()
+                    .credentialsProvider(credentialsProvider)
+                    .region(connectionConfig.region()).build();
+            bAwsMpeClient.addNativeData(NATIVE_CLIENT, nativeClient);
+        } catch (Exception e) {
+            String errorMsg = String.format("Error occurred while initializing the marketplace entitlement client: %s",
+                    e.getMessage());
+            return CommonUtils.createError(errorMsg, e);
+        }
+        return null;
+    }
+
+    private static AwsCredentials getCredentials(ConnectionConfig connectionConfig) {
+        if (Objects.nonNull(connectionConfig.sessionToken())) {
+            return AwsSessionCredentials.create(connectionConfig.accessKeyId(), connectionConfig.secretAccessKey(),
+                    connectionConfig.sessionToken());
+        } else {
+            return AwsBasicCredentials.create(connectionConfig.accessKeyId(), connectionConfig.secretAccessKey());
+        }
+    }
```

Comment:
Can't we move this responsibility inside `ConnectionConfig` and make it as class?

Reply:
`ConnectionConfig` maps directly into the Ballerina `ConnectionConfig` record. Hence, -1 for that.